src/legacy/ui/public/filter_bar/query_filter.js

```diff
@@ -75,19 +85,14 @@ export function FilterBarQueryFilterProvider(Promise, indexPatterns, $rootScope,
    * @param {bool} global Whether the filter should be added to global state
    * @returns {Promise} filter map promise
    */
-  queryFilter.addFilters = function (filters, global) {
-
-    if (global === undefined) {
-      const configDefault = config.get('filters:pinnedByDefault');
-
-      if (configDefault === false || configDefault === true) {
```

Comment:
Looks like this check used to confirm we are only assigning a boolean value to `global`.

Are we positive there are no circumstances under which `pinnedByDefault` would not be boolean? (I'm assuming the answer is yes). Only asking because otherwise you could get an unintentional truthy in the ternary on line 95.

Reply:
if its required, can we change it to `if (typeof configDefault === 'boolean')`

Reply:
I agree with this observation, but there are no signs in the code for it being non-boolean, especially this code being more that 3 years old.

I prefer to define the correct behavior (which I understand to be: "If `addToGlobalState` is not specified, use the default from configs").